examples/http/azure_network_interfaces.js

```diff
@@ -0,0 +1,571 @@
+        }
+    }, {
+        label: "Disable TCP State Tracking",
+        valueType: D.valueType.STRING,
+        key: "disableTcpStateTracking",
+        extract: function (networkInterface) {
+            return networkInterface.properties && networkInterface.properties.disableTcpStateTracking || "N/A"
+        }
+    }, {
+        label: "Network Security Group",
+        valueType: D.valueType.STRING,
+        key: "networkSecurityGroup",
+        extract: function (networkInterface) {
+            if (networkInterface.properties && networkInterface.properties.networkSecurityGroup && networkInterface.properties.networkSecurityGroup.id) {
+                const matches = networkInterface.properties.networkSecurityGroup.id.match(/networkSecurityGroups\/([^/]+)/)
+                return matches ? matches[1] : "N/A"
+            }
+            return "N/A"
+        }
+    }, {
+        label: "Primary Network Interface",
+        valueType: D.valueType.STRING,
+        key: "primaryNetworkInterface",
+        extract: function (networkInterface) {
+            return networkInterface.properties && networkInterface.properties.primary || "N/A"
+        }
+    }, {
+        label: "Associated Virtual Machine",
+        valueType: D.valueType.STRING,
+        key: "associatedVirtualMachine",
+        extract: function (networkInterface) {
+            if (networkInterface.properties && networkInterface.properties.virtualMachine && networkInterface.properties.virtualMachine.id) {
+                const matches = networkInterface.properties.virtualMachine.id.match(/virtualMachines\/([^/]+)/)
+                return matches ? matches[1] : "N/A"
+            }
+            return "N/A"
+        }
+    }, {
+        label: "NIC Type", valueType: D.valueType.STRING, key: "nicType", extract: function (networkInterface) {
+            return networkInterface.properties && networkInterface.properties.nicType || "N/A"
+        }
+    }, {
+        label: "Location", valueType: D.valueType.STRING, key: "location", extract: function (networkInterface) {
+            return networkInterface.location || "N/A"
+        }
+    }, {
+        label: "Kind", valueType: D.valueType.STRING, key: "kind", extract: function (networkInterface) {
+            return networkInterface.kind || "N/A"
+        }
+    }
+]
+
+/**
+ * Generates network interface properties by extracting information from the defined virtualMachineScaleSetInfoExtractors.
+ * @returns {Promise} A promise that resolves when network interface properties are generated.
+ * It populates the global variable `virtualMachineScaleSetProperties` and concatenates them with `networkInterfaceInfoExtractors` and `performanceMetrics`.
+ */
+function generateNetworkInterfaceProperties() {
+    return D.q.all(networkInterfaceInfoExtractors.map(function (extractorInfo) {
+        return new Promise(function (resolve) {
```

Comment:
why you are using promise instead of returning a simple result?

do not use Promise library for promises, use D.q

you can start filtring your properties with !== 'id' then you map the result and concat with the other columns

the function could be simple like this
function generateNetworkInterfaceProperties() {
    return networkInterfaceInfoExtractors.filter(function (extractorInfo) {
            return extractorInfo.key !== 'id'
        }).map(function (extractorInfo) {
             return {
                    'key': extractorInfo.key,
                    'label': extractorInfo.label,
                    'valueType': extractorInfo.valueType,
                    'unit': extractorInfo.unit ? extractorInfo.unit : null
                }
        }).concat(performanceMetrics)
}